src/averageAges.js

```diff
@@ -34,7 +43,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const FEMALE = 'f';
+  function isMother(people, mother) {
+    return people.some((person) => {
+      return person.mother === mother;
+    });
+  }
+  const women = people.filter((person) => {
+    return withChildren ? (person.sex === FEMALE && isMother(people, person.name)) : person.sex === FEMALE;
```

Comment:
when using `isMother` complexity becomes O(n^2) (a loop inside a loop). But it may be solved with O(n) complexity. If you want, you could try to create a hash map and `isMother` operation will be cost O(1) instead of O(n)
https://www.youtube.com/watch?v=v4cd1O4zkGw
http://bigocheatsheet.com/